src/tools/rustfmt/src/utils.rs

```diff
@@ -74,6 +74,35 @@ pub(crate) fn format_visibility(
     }
 }
 
+// Does not allocate in the common, implied case.
+pub(crate) fn format_restriction<Kind: ast::RestrictionKind>(
```

Comment:
Not yet, but on the agenda/backlog

Reply:
Know there's been some discussion about this on Zulip already, but posting inline for awareness.

First I just want to say thank you for incorporating formatting and various tests as part of this :pray: We discussed this at the most recent T-Style meeting, and we do anticipate this will most likely be the formatting we'll want to eventually utilize. However, we've still got a couple process questions to iron out relative to style/formatting for new syntax constructs, so in the interim we'd like to see the rustfmt changes reverted and instead apply the historical default behavior where rustfmt will just re-emit whatever the user wrote.

Ultimately up to you how you'd like to do that, but certainly feel free to leave the new `format_restriction` function in place (with a corresponding attr to allow unused) to make it easier for t-rustfmt to flip it on later. Then let's update the respective call sites to instead emit the original contents of the trait/field def (e.g. `context.snippet(span).to_owned())`) if the restriction is anything other than implied